lib/Controller/SocialApiController.php

```diff
@@ -0,0 +1,188 @@
+		if (!is_array($socialentry)) {
+			throw new Exception("socialentry format missmatch"); // FIXME: the Exceptions seem not to work as expected...
+		}
+
+		$candidate = $socialentry[3];
+		$network   = $socialentry[1]['type'];
+		$connector = null;
+
+		if (is_array($network)) { $network = $network[0]; }
+
+		// get profile-id
+		switch ($network) {
+			case "facebook":
+				$candidate = basename($candidate);
+				if (!ctype_digit($candidate)) {
+					// TODO: determine facebook profile id from username
+					throw new Exception("facebook profile-id expected to be a number, not %s", $candidate);
+				}
+				break;
+			default:
+				throw new Exception("%s not implemented", $network);
+		}
+
+		// build connector
+		switch ($network) {
+			case "facebook":
+				switch ($type) {
+					case "avatar":
+						$connector = "https://graph.facebook.com/" . ($candidate) . "/picture?width=720";
+						break;
+					default:
+						throw new Exception("%s for %s not implemented", $type, $network);
+				}
+				break;
+			default:
+				throw new Exception("Unexpected error building the connector for %s", $network);
+		}
+
+		return ($connector);
+
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * Retrieves social profile data for a contact
+	 *
+	 * @param string addressbookId identifier of the addressbook
+	 * @param string contact identifier of the contact
+	 * @param string type which information to get: avatar, ...
+	 */
+	public function fetch($addressbookId, $contactId, $type) {
+
+		$url = null;
+		$response = 404;
+
+		try {
+			// get social parameters from contact
+
+			/* TODO - port from javascript...
+			const jCal = this.contact.jCal
+			const socialentries = jCal[1].filter(props => props[0] === 'x-socialprofile')
+			socialentries.forEach(getSocialConnector)
+			*/
+			// FIXME static for testing:
+			$socialprofile = array("foo",array("type" => "facebook"),"bar","https://facebook.com/4");
+
+			try {
+				$url = $this->getSocialConnector($socialprofile, $type);
+			}
+			catch (Exception $e) {
+				$response = 500;
+				throw new Exception($e->getMessage());
+			}
+
+			if (empty($url)) {
+				$response = 500;
+				throw new Exception('not implemented');
+			}
+
+			$host = parse_url($url);
+			if (!$host) {
+				$response = 404;
+				throw new Exception('Could not parse URL');
+			}
+			$opts = [
+				"http" => [
+					"method" => "GET",
+					"header" => "User-Agent: Nextcloud Contacts App"
+				]
+			];
+			$context = stream_context_create($opts);
+			$socialdata = file_get_contents($url, false, $context);
+			if (!$socialdata) {
+				$response = 404;
+				throw new Exception('Could not parse URL');
+			}
+
+			$response = 200;
+
+			switch ($type) {
+				case "avatar":
+					header("Content-type:image/png");
+					break;
+				default:
+					header("Content-type:application/json");
+			}
+			
+			echo $socialdata;
+		} 
+		catch (Exception $e) {
+		}
+
+		http_response_code($response);
+		exit;
+
```

Comment:
Please properly return a proper Response here :)
https://github.com/nextcloud/server/blob/caff1023ea72bb2ea94130e18a2a6e2ccf819e5f/lib/public/AppFramework/Http/JSONResponse.php

Reply:
or the appropriate response, not forced to be a JSONResponse of course, pick accordingly